internal/builder/builder.go

```diff
@@ -393,10 +457,18 @@ func (b *Builder) Save(logger logging.Logger, creatorMetadata CreatorMetadata) e
 	if _, err := dist.GetLabel(b.image, dist.ExtensionLayersLabel, &extLayers); err != nil {
 		return errors.Wrapf(err, "getting label %s", dist.ExtensionLayersLabel)
 	}
-	err = b.addModules(buildpack.KindExtension, logger, tmpDir, b.image, b.additionalExtensions, extLayers)
+
+	var excludedExtensions []buildpack.BuildModule
+	excludedExtensions, err = b.addFlattenModules(buildpack.KindExtension, logger, tmpDir, b.image, b.additionalExtensions.FlattenModules(), extLayers)
+	if err != nil {
+		return err
+	}
+
+	err = b.addModules(buildpack.KindExtension, logger, tmpDir, b.image, append(b.additionalExtensions.NonFlattenModules(), excludedExtensions...), extLayers)
```

Comment:
```suggestion
	err = b.addExplodedModules(buildpack.KindExtension, logger, tmpDir, b.image, append(b.additionalExtensions.NonFlattenModules(), excludedExtensions...), extLayers)
```
(or something, whatever is consistent)